topics/admin/tutorials/job-destinations/tutorial.md

```diff
@@ -212,100 +388,122 @@ We want our tool to run with more than one core. To do this, we need to instruct
 
-> <hands-on-title>Writing a dynamic job destination</hands-on-title>
+> <hands-on-title>Configuring defaults and inheritance</hands-on-title>
 >
-> 1. Create and open `files/galaxy/dynamic_job_rules/my_rules.py`
+> 1. Edit your `files/galaxy/config/tpv_rules_local.yml` and add the following settings.
 >
 >    {% raw %}
 >    ```diff
->    --- /dev/null
->    +++ b/files/galaxy/dynamic_job_rules/my_rules.py
->    @@ -0,0 +1,10 @@
->    +from galaxy.jobs import JobDestination
->    +from galaxy.jobs.mapper import JobMappingException
->    +import os
+>    --- a/files/galaxy/config/tpv_rules_local.yml
+>    +++ b/files/galaxy/config/tpv_rules_local.yml
+>    @@ -1,4 +1,11 @@
+>    +global:
+>    +  default_inherits: default
 >    +
->    +def admin_only(app, user_email):
->    +    # Only allow the tool to be executed if the user is an admin
->    +    admin_users = app.config.get( "admin_users", "" ).split( "," )
->    +    if user_email not in admin_users:
->    +        raise JobMappingException("Unauthorized.")
->    +    return JobDestination(runner="slurm")
+>     tools:
+>    +  default:
+>    +    abstract: true
+>    +    cores: 1
+>    +    mem: cores * 4
+>       .*testing.*:
+>         cores: 2
+>         mem: cores * 4
+>    @@ -27,4 +34,3 @@ destinations:
+>         max_accepted_cores: 16
+>         params:
+>           native_specification: --nodes=1 --ntasks=1 --cpus-per-task={cores}
+>    -
 >    {% endraw %}
 >    ```
->    {: data-commit="Add my rules python script"}
+>    {: data-commit="Add TPV default inherits"}
 >
->    This destination will check that the `user_email` is in the set of `admin_users` from your config file.
+>    We have defined a `global` section specifying that all tools and destinations should inherit from a specified `default`. We have then defined a tool named `default`, whose properties
+>    are implicitly inherited by all tools at runtime. This means that our `testing` tool will also inherit from this default tool, but it explicitly overrides cores
+>    We can also explicitly specify an `inherits` clause if we wish to extend a specific tool or destination, as previously shown in the destinations section.
 >
->    > <tip-title>Debugging dynamic destinations</tip-title>
->    > You can use `pdb` for more advanced debugging, but it requires some configuration. `print()` statements are usually sufficient and easier.
->    {: .tip}
+> 2. Run the Galaxy playbook. When the new `tpv_rules_local.yml` is copied, TPV will automatically pickup the changes without requiring a restart of Galaxy.
```

Comment:
Is this true? I don't think we have set `watch_job_rules` in the config, which I think is what TPV uses to enable watching.